base/array.jl

```diff
@@ -1054,16 +1054,23 @@ function lexcmp(a::Array{UInt8,1}, b::Array{UInt8,1})
 end
 
 # use memcmp for == on bit integer types
-function ==(a::Array{T,N}, b::Array{T,N}) where T<:BitInteger where N
+
+function ==(a::A, b::A) where A <: Union{Array{Int8,N},Array{UInt8,N},Array{Int16,N},
+                                         Array{UInt16,N},Array{Int32,N},Array{UInt32,N},
+                                         Array{Int64,N},Array{UInt64,N},Array{Int128,N},
+                                         Array{UInt128,N}} where N
```

Comment:
You can use something like `const BitIntegerArray = Union{map(T->Array{T}, uniontypes(BitInteger))...}`.